src/navbar/navbar-items.tsx

```diff
@@ -43,25 +44,15 @@ export const NavbarItems = <T,>({
     // =============================================================================
     // EFFECTS
     // =============================================================================
-    useEffect(() => {
-        const handleClickOutside = (event: MouseEvent) => {
-            if (ref.current && !ref.current.contains(event.target as Node)) {
-                onBlur();
-            }
-        };
-        document.addEventListener("click", handleClickOutside, true);
-        return () => {
-            document.removeEventListener("click", handleClickOutside, true);
-        };
-    }, []);
+    const handleBlur = () => {
+        setShowSubMenu(false);
+    };
+
+    useBlur(ref, handleBlur);
```

Comment:
refactor this useEffect to util as it is used for both avatar and navbar (this hook is to handle the "blur" behaviour to dismiss the submenu if user clicks outside of the component)